src/modules/Update.rb

```diff
@@ -222,24 +223,15 @@ def IsProductSupportedForUpgrade
     #-----------------------------------------------------------------------
 
     def SelectedProducts
-      selected = Pkg.ResolvableProperties("", :product, "")
-      selected = Builtins.filter(selected) do |p|
-        Ops.get(p, "status") == :selected
-      end
-      Builtins.maplist(selected) do |p|
-        Ops.get_locale(
-          p,
-          "display_name",
-          Ops.get_locale(
-            p,
-            "summary",
-            Ops.get_locale(
-              p,
-              "name",
-              Ops.get_locale(p, "version", _("Unknown Product"))
-            )
-          )
-        )
+      selected = Y2Packager::Resolvable.find(kind: :product, status: :selected)
+
+      selected.map do |p|
+        next p.display_name unless p.display_name.empty?
+        next p.summary unless p.summary.empty?
+        next p.name unless p.name.empty?
+        next p.version unless p.version.empty?
```

Comment:
NP: I'd probably skip the version, just the number does not tell much and might be misleading. Moreover the `name` should really not be empty, that would mean the product is quite broken by definition. The `version` branch should not be normally reached.